cpp/lib/src/DbConnection.cc

```diff
@@ -449,6 +449,61 @@ void DbConnection::insertIntoTableOrDie(const std::string &table_name,
 }
 
 
+void DbConnection::insertIntoTableOrDie(const std::string &table_name, const std::vector<std::string> &column_names,
+                                        const std::vector<std::vector<std::string>> &values,
+                                        const DuplicateKeyBehaviour duplicate_key_behaviour, const std::string &where_clause)
+{
+    if (column_names.empty())
+        LOG_ERROR("at least one column name must be provided!");
+
+    if (not where_clause.empty() and duplicate_key_behaviour != DKB_REPLACE)
+        LOG_ERROR("\"where_clause\" is only valid when using the DKB_REPLACE mode!");
+
+    std::string insert_stmt(duplicate_key_behaviour == DKB_REPLACE ? "REPLACE" : "INSERT");
+    if (duplicate_key_behaviour == DKB_IGNORE)
+        insert_stmt += (type_ == T_MYSQL) ? " IGNORE" : " OR IGNORE";
+    insert_stmt += " INTO " + table_name + " (";
+
+    const char column_name_quote(type_ == T_MYSQL ? '`' : '"');
+
+    bool first(true);
+    for (const auto &column_name : column_names) {
+        if (not first)
+            insert_stmt += ',';
+        else
+            first = false;
+
+        insert_stmt += column_name_quote;
+        insert_stmt += column_name;
+        insert_stmt += column_name_quote;
+    }
+
+    insert_stmt += ") VALUES ";
+
+    for (auto column_values(values.cbegin()); column_values != values.cend(); ++column_values) {
+        insert_stmt += '(';
+        first = true;
+        for (const auto &column_value : *column_values) {
+            if (not first)
+                insert_stmt += ',';
+            else
+                first = false;
+
+            insert_stmt += escapeAndQuoteString(column_value);
+        }
+        insert_stmt += ')';
+
+        if (column_values != values.cend() - 1)
+            insert_stmt += ',';
+    }
+
```

Comment:
Wouldn't StringUtil::Join after escapeAndQuoteString on the elements be a more straightforward approach?

Reply:
It would require two additional temporary vector.  The first, let's call it quoted_values and the second one would be the one generated by StringUtil::Join.  I'd rather not do that.

Reply:
Ok, but doesn't StringUtil::Join() only assemble the vector argument to a string?